src/ui/components/CreateIdentifier/components/IdentifierStage1.tsx

```diff
@@ -87,8 +88,12 @@ const IdentifierStage1 = ({
   }, [groupMetadata, currentOperation, groupId, multiSigGroupCache]);
 
   useEffect(() => {
-    incomingRequest.multisigIcpDetails?.ourIdentifier.groupMetadata?.groupId ===
-      groupId && handleDone();
+    if (
+      incomingRequest?.type === IncomingRequestType.MULTI_SIG_REQUEST_INCOMING
+    ) {
+      incomingRequest.multisigIcpDetails?.ourIdentifier.groupMetadata
```

Comment:
We can lose the `?` here since it's now guaranteed to be set (helps us catch if we accidentally made it optional, or added another type with `IncomingRequestType.MULTI_SIG_REQUEST_INCOMING` by accident)

Reply:
but `incomingRequest` can be `undefined`

Reply:
I mean the `?` after `multisigIcpDetails`